src/pal/pal_linux.h

```diff
@@ -85,6 +94,42 @@ namespace snmalloc
         madvise(p, size, MADV_FREE);
       }
     }
+
+    static uint64_t get_entropy64()
+    {
+#  ifdef SNMALLOC_PLATFORM_HAS_GETENTROPY
+      uint64_t result;
+      if (getentropy(&result, sizeof(result)) != 0)
+        error("Failed to get system randomness");
```

Comment:
Not a new bug, but this can cause deadlocks during early init.  If the system call fails then the POSIX PAL calls libc functions that can require malloc.  We should modify the implementation of `error` to use `write`.

```suggestion
      {
        error("Failed to get system randomness");
      }
```